### PR TITLE
Wasmvm

### PR DESCRIPTION
**Pull request by BugFreeSoftware**
Initial client code generation for TypeScript and Go

### CODE REVIEW

File: client/callview.go [hunk 1]
```diff
@@ -28,7 +28,7 @@ func (c *WaspClient) CallView(chainID *iscp.ChainID, hContract iscp.Hname, funct
 	var res dict.Dict
 	var err error
 	for {
-		err = c.do(http.MethodGet, routes.CallView(chainID.Base58(), hContract.String(), functionName), arguments, &res)
+		err = c.do(http.MethodPost, routes.CallView(chainID.Base58(), hContract.String(), functionName), arguments, &res)
```
Comment (kape1395) [inline client/callview.go:31]: Why a query requires the POST method?

File: contracts/wasm/erc721/src/state.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ use crate::*;
 use crate::keys::*;
 use crate::typedefs::*;
 
+#[derive(Clone, Copy)]
```
Comment (kape1395) [inline contracts/wasm/erc721/src/state.rs:18]: I guess a lot of these changes are in the generated files?
Maybe we can mark them as generated in git/github?
https://docs.github.com/en/repositories/working-with-files/managing-files/customizing-how-changed-files-appear-on-github

File: contracts/wasm/testwasmlib/ts/testwasmlib/contract.ts [hunk 3]
```diff
@@ -14,6 +14,7 @@ export class ArrayClearCall {
 }
 
 export class ArrayClearContext {
+	events:  sc.TestWasmLibEvents = new sc.TestWasmLibEvents();
```
Comment (kape1395) [inline contracts/wasm/testwasmlib/ts/testwasmlib/contract.ts]: Double space after `events:` in multiple places. That's minor, of course.

File: packages/vm/wasmlib/ts/wasmclient/arguments.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+import * as wasmclient from "./index"
+import {Buffer} from "./buffer";
+
+// The Arguments struct is used to gather all arguments for this smart
+// contract function call and encode it into this deterministic byte array
+export class Arguments extends wasmclient.Encoder {
+    private args = new Map<string, wasmclient.Bytes>();
+
+    // get(key: string): wasmclient.Bytes {
```
Comment (kape1395) [inline packages/vm/wasmlib/ts/wasmclient/arguments.ts:12]: Clean-up it?

File: packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+export class GetEventsForRequestView extends wasmclient.ClientView {
+	private args: wasmclient.Arguments = new wasmclient.Arguments();
+	
+	public requestID(v: wasmclient.RequestID): void {
+		this.args.set(ArgRequestID, this.args.fromRequestID(v));
+	}
+
+	public async call(): Promise<GetEventsForRequestResults> {
+		this.args.mandatory(ArgRequestID);
+        const res = new GetEventsForRequestResults();
```
Comment (kape1395) [inline packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts]: This const is misalligned in multiple places.

**Review by jorgemmsilva — APPROVED** [review]

**Review by kape1395 — APPROVED** [review]
Just a few minor notes.

File: client/callview.go (hunk 1 above)
Comment (BugFreeSoftware) [reply client/callview.go:31]: This was part of a test, to see if POST works as well. On the TypeScript side fetch() would not allow a body to be passed to a GET request. Changing it to POST made it run. I checked the web API and it implements identical routes for GET and POST. To have both Go and TypeScript client code work identical I let the change stand on the Go side.

File: contracts/wasm/testwasmlib/ts/testwasmlib/contract.ts (hunk 3 above)
Comment (BugFreeSoftware) [reply contracts/wasm/testwasmlib/ts/testwasmlib/contract.ts]: Fixed

File: packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts (hunk 5 above)
Comment (BugFreeSoftware) [reply packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts]: Fixed

File: contracts/wasm/erc721/src/state.rs (hunk 2 above)
Comment (BugFreeSoftware) [reply contracts/wasm/erc721/src/state.rs:18]: That's a good find. Would be a good solution for now.

File: packages/vm/wasmlib/ts/wasmclient/arguments.ts (hunk 4 above)
Comment (BugFreeSoftware) [reply packages/vm/wasmlib/ts/wasmclient/arguments.ts:12]: Still in the process of cleaning up. Just wanted to merge to develop as soon as possible.

File: packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts (hunk 5 above)
Comment (BugFreeSoftware) [reply packages/vm/wasmlib/ts/wasmclient/coreblocklog/service.ts:143]: Fixed

**Pull request merged**